File: java/eyes.playwright.java/src/main/java/com/applitools/eyes/playwright/universal/driver/dto/DriverInfoDto.java
package com.applitools.eyes.playwright.universal.driver.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.Objects;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DriverInfoDto {

    private DriverInfoFeaturesDto features;

    public DriverInfoDto() {
        this.features = new DriverInfoFeaturesDto();
    }

    public DriverInfoFeaturesDto getFeatures() {
        return features;
    }

    public void setFeatures(DriverInfoFeaturesDto features) {
        this.features = features;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DriverInfoDto that = (DriverInfoDto) o;
        return Objects.equals(features, that.features);
    }

    @Override
    public int hashCode() {
        return Objects.hash(features);
    }

    @Override
    public String toString() {
        return "DriverInfoDto{" +
                "features=" + features +
                '}';
    }
}

File: dotnet/Eyes.Playwright.DotNet/Universal/Driver/Dto/DriverInfoDto.cs
using Newtonsoft.Json;

namespace Applitools.Playwright.Universal.Driver.Dto
{
    public class DriverInfoDto
    {
        [JsonProperty("features")]
        public DriverInfoFeaturesDto Features { get; set; }

        public DriverInfoDto()
        {
            Features = new DriverInfoFeaturesDto();
        }

        public override bool Equals(object o)
        {
            if (this == o)
            {
                return true;
            }

            if (!(o is DriverInfoDto other))
            {
                return false;
            }

            return Features.Equals(other.Features);
        }

        public override int GetHashCode()
        {
            return Features?.GetHashCode() ?? 0;
        }

        public override string ToString()
        {
            return "DriverInfoDto{" +
                   "features=" + Features +
                   '}';
        }
    }
}